components/range_cache_memory_engine/src/background.rs

```diff
@@ -656,14 +660,27 @@ impl BackgroundRunnerCore {
                 assert!(tikv_util::thread_group::is_shutdown(!cfg!(test)));
             }
         }
-        let mut regions_map = self.engine.region_manager().regions_map.write();
-        for region in regions_to_load {
-            let cache_region = CacheRegion::from_region(&region);
-            if let Err(e) = regions_map.load_region(cache_region) {
-                error!("ime error loading range"; "cache_range" => ?region, "err" => ?e);
+        for _ in 0..evict_count {
+            if rx.recv().await.is_none() {
+                break;
+            }
+        }
+        if !self.memory_controller.reached_stop_load_limit() {
+            let expected_new_count = (self
+                .memory_controller
+                .soft_limit_threshold()
+                .saturating_sub(self.memory_controller.mem_usage()))
+                / range_stats_manager.expected_region_size;
+            let expected_new_count = usize::max(expected_new_count, 1);
+            let mut regions_map = self.engine.region_manager().regions_map.write();
+            for region in regions_to_load.into_iter().take(expected_new_count) {
+                let cache_region = CacheRegion::from_region(&region);
+                if let Err(e) = regions_map.load_region(cache_region) {
+                    error!("ime error loading range"; "cache_range" => ?region, "err" => ?e);
```

Comment:
```suggestion
                    warn!("ime error loading range"; "cache_range" => ?region, "err" => ?e);
```
Return error means range or id conflict, so it is an expected behavior, thus warn level is enough.